pkg/encryption/encryption.go

```diff
@@ -33,78 +33,86 @@ func Increment(nonce *storj.Nonce, amount int64) (truncated bool, err error) {
-func Encrypt(data []byte, cipher storj.Cipher, key *storj.Key, nonce *storj.Nonce) (cipherData []byte, err error) {
+func Encrypt(data []byte, cipher storj.CipherSuite, key *storj.Key, nonce *storj.Nonce) (cipherData []byte, err error) {
 	// Don't encrypt empty slice
 	if len(data) == 0 {
 		return []byte{}, nil
 	}
 
 	switch cipher {
-	case storj.Unencrypted:
+	case storj.EncUnspecified:
 		return data, nil
-	case storj.AESGCM:
+	case storj.EncNull:
+		return data, nil
+	case storj.EncAESGCM:
 		return EncryptAESGCM(data, key, ToAESGCMNonce(nonce))
-	case storj.SecretBox:
+	case storj.EncSecretBox:
 		return EncryptSecretBox(data, key, nonce)
 	default:
 		return nil, ErrInvalidConfig.New("encryption type %d is not supported", cipher)
 	}
 }
 
 // Decrypt decrypts cipherData with the given cipher, key and nonce
-func Decrypt(cipherData []byte, cipher storj.Cipher, key *storj.Key, nonce *storj.Nonce) (data []byte, err error) {
+func Decrypt(cipherData []byte, cipher storj.CipherSuite, key *storj.Key, nonce *storj.Nonce) (data []byte, err error) {
 	// Don't decrypt empty slice
 	if len(cipherData) == 0 {
 		return []byte{}, nil
 	}
 
 	switch cipher {
-	case storj.Unencrypted:
+	case storj.EncUnspecified:
+		return cipherData, nil
+	case storj.EncNull:
 		return cipherData, nil
-	case storj.AESGCM:
+	case storj.EncAESGCM:
 		return DecryptAESGCM(cipherData, key, ToAESGCMNonce(nonce))
-	case storj.SecretBox:
+	case storj.EncSecretBox:
 		return DecryptSecretBox(cipherData, key, nonce)
 	default:
 		return nil, ErrInvalidConfig.New("encryption type %d is not supported", cipher)
 	}
 }
 
 // NewEncrypter creates a Transformer using the given cipher, key and nonce to encrypt data passing through it
-func NewEncrypter(cipher storj.Cipher, key *storj.Key, startingNonce *storj.Nonce, encryptedBlockSize int) (Transformer, error) {
+func NewEncrypter(cipher storj.CipherSuite, key *storj.Key, startingNonce *storj.Nonce, encryptedBlockSize int) (Transformer, error) {
 	switch cipher {
-	case storj.Unencrypted:
+	case storj.EncUnspecified:
 		return &NoopTransformer{}, nil
-	case storj.AESGCM:
+	case storj.EncNull:
+		return &NoopTransformer{}, nil
+	case storj.EncAESGCM:
 		return NewAESGCMEncrypter(key, ToAESGCMNonce(startingNonce), encryptedBlockSize)
-	case storj.SecretBox:
+	case storj.EncSecretBox:
 		return NewSecretboxEncrypter(key, startingNonce, encryptedBlockSize)
 	default:
 		return nil, ErrInvalidConfig.New("encryption type %d is not supported", cipher)
 	}
 }
 
 // NewDecrypter creates a Transformer using the given cipher, key and nonce to decrypt data passing through it
-func NewDecrypter(cipher storj.Cipher, key *storj.Key, startingNonce *storj.Nonce, encryptedBlockSize int) (Transformer, error) {
+func NewDecrypter(cipher storj.CipherSuite, key *storj.Key, startingNonce *storj.Nonce, encryptedBlockSize int) (Transformer, error) {
 	switch cipher {
-	case storj.Unencrypted:
+	case storj.EncUnspecified:
```

Comment:
This is not the correct behavior, it should return an error here.

Reply:
updated